sql/core/src/main/scala/org/apache/spark/sql/ExperimentalMethods.scala

```diff
@@ -44,11 +44,14 @@ class ExperimentalMethods private[sql]() {
    */
   @volatile var extraStrategies: Seq[Strategy] = Nil
 
+  @volatile var extraPreOptimizations: Seq[Rule[LogicalPlan]] = Nil
+
   @volatile var extraOptimizations: Seq[Rule[LogicalPlan]] = Nil
```

Comment:
how about rename this `extraPostOptimizations`?

Reply:
This is an API change. We can't do it.

Reply:
Yes, i agree with @gatorsmile, renaming `extraOptimizations ` to `extraPostOptimizations` will be symmetric with `extraPreOptimizations`, but doing so may affect the existing API calls.